tests/testsuite/build.rs

```diff
@@ -815,13 +824,20 @@ fn cargo_compile_with_invalid_code() {
 
     p.cargo("build")
         .with_status(101)
-        .with_stderr_contains(
-            "[ERROR] could not compile `foo` (bin \"foo\") due to 1 previous error\n",
-        )
+        .with_stderr_data(str![[r#"
+[WARNING] path `src/foo.rs` was erroneously implicitly accepted for binary `foo`,
+please set bin.path in Cargo.toml
+[COMPILING] foo v0.5.0 ([ROOT]/foo)
+[ERROR] [..]
+...
+[ERROR] could not compile `foo` (bin "foo") due to 1 previous error
+
+"#]])
         .run();
     assert!(p.root().join("Cargo.lock").is_file());
 }
 
+#[allow(deprecated)]
```

Comment:
Does this need the deprecated functions?

Reply:
Updated to snapbox. It was multiple `...` and `unordered` that made snapbox confused. I figured it out then.